app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationActions.jsx

```diff
@@ -85,7 +85,30 @@ export default function RegistrationActions({
         },
       },
     );
-  };
+  }, [competitionInfo.id]);
+
+  const moveToWaitingList = useCallback((attendees) => {
+    const registrationsByUserId = _.groupBy(registrations, 'user_id');
```

Comment:
actually, now that you have this. You should be able to skip computing `attendeesWithRegistrations` at all and just use `registrationsByUserId` like 
    const [paid, unpaid] = _.partition(attendees, ({ user_id }) => registrationsByUserId[user_id].registration?.payment?.updated_at);

Reply:
Done!